src/app/examples/crud/books.effects.spec.ts

```diff
@@ -1,77 +1,72 @@
-    entities: {
-      '1': {
-        author: 'Author',
-        description: 'Description',
-        id: '1',
-        title: 'Title'
-      }
-    },
-    ids: ['1']
-  };
+  describe('persistBooks', () => {
+    const booksState: BookState = {
+      entities: {
+        '1': {
+          author: 'Author',
+          description: 'Description',
+          id: '1',
+          title: 'Title'
+        }
+      },
+      ids: ['1']
+    };
+    let localStorage: LocalStorageService;
+    let store: Store<any>;
 
-  beforeEach(() => {
-    TestBed.configureTestingModule({
-      imports: [StoreModule.forRoot({})],
-      providers: [
-        BooksEffects,
-        provideMockActions(() => actions$),
-        provideMockStore(),
-        {
-          provide: LocalStorageService,
-          useValue: jasmine.createSpyObj('LocalStorageService', ['setItem'])
+    beforeEach(() => {
+      localStorage = jasmine.createSpyObj('localStorage', ['setItem']);
+      store = of({
+        examples: {
+          books: booksState
         }
-      ]
+      }) as any;
     });
-    localStorageService = TestBed.get(LocalStorageService);
-    bookEffects = TestBed.get(BooksEffects);
-    store = TestBed.get(Store);
-    state = createState(booksState);
-    store.setState(state);
-  });
 
-  it('should be created', () => {
-    expect(bookEffects).toBeTruthy();
-  });
+    it('should not dispatch any actions', () => {
+      const actions = new Actions(EMPTY);
+      const effects = new BooksEffects(actions, store, localStorage);
+      const metadata = getEffectsMetadata(effects);
 
-  it('persistBooks should not dispatch any action', () => {
-    metadata = getEffectsMetadata(bookEffects);
-    expect(metadata.persistBooks).toEqual({ dispatch: false });
-  });
+      expect(metadata.persistBooks).toEqual({ dispatch: false });
+    });
+
+    it('should call setItem on LocalStorageService for delete one action', () => {
+      const action = new ActionBooksDeleteOne({ id: '1' });
+      const source = cold('a|', { a: action });
+      const actions = new Actions(source);
+      const effects = new BooksEffects(actions, store, localStorage);
+
+      effects.persistBooks.subscribe(() => {
+        expect(localStorage.setItem).toHaveBeenCalledWith(
+          BOOKS_KEY,
+          booksState
+        );
+      });
+    });
+
+    it('should call setItem on LocalStorageService for upsert one action', () => {
+      const action = new ActionBooksUpsertOne({ book: {} as any });
+      const source = cold('a|', { a: action });
+      const actions = new Actions(source);
+      const effects = new BooksEffects(actions, store, localStorage);
 
-  it('should call setItem on LocalStorageService for add one action', () => {
-    bookEffects.persistBooks.subscribe(() => {
-      expect(localStorageService.setItem).toHaveBeenCalledWith(
-        BOOKS_KEY,
-        state
-      );
+      effects.persistBooks.pipe(take(1)).subscribe(() => {
```

Comment:
I think the `take(1)` operator isn't needed here? Or is it?

Reply:
It is not. I'll remove it :)